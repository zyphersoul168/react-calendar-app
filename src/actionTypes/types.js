export const types = {
  uiOpenModal: "[ui] open modal",
  uiCloseModal: "[ui] close modal",

  eventSetActive: "[event] set active",
  eventLogout: "[event] logout",
  eventStartAddNew: "[event] start add new",
  eventAddNew: "[event] add new",
  eventClearActive: "[event] clear active",
  eventUpdated: "[event] updated",
  eventDeleted: "[event] deleted",
  eventLoad: "[event] get events",

  // authChecking: "[auth] checking login state",
  authCheckingFinish: "[auth] finish checking login state",
  authStartLogin: "[auth] start login",
  authLogin: "[auth] login",
  authStartSignup: "[auth] start register",
  authStartTokenRevalidate: "[auth] start revalidate token",
  authLogout: "[auth] logout",
};
