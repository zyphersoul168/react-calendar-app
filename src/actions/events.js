import Swal from "sweetalert2";
import { types } from "../actionTypes/types";
import { fetchWithToken } from "../helpers/fetch";
import { parseEvents } from "../helpers/parseEvents";

// async actions
export const eventStartAddNew = (event) => {
  return async (dispatch, getState) => {
    const { uid, name } = getState().auth;

    try {
      const resp = await fetchWithToken("events", event, "POST");
      const body = await resp.json();

      if (body.ok) {
        event._id = body.event._id;
        event.user = {
          _id: uid,
          name: name,
        };
        console.log(event);
        dispatch(eventAddNew(event));
      }
    } catch (error) {
      console.log(error);
    }
  };
};

export const eventStartLoading = () => {
  return async (dispatch) => {
    try {
      const resp = await fetchWithToken("events");
      const body = await resp.json();
      const events = parseEvents(body.events);
      dispatch(eventLoad(events));
      
    } catch (error) {
      console.log(error);
    }
  };
};

export const eventStartUpdate = (event) => {
  return async(dispatch) => {
    try {
      const resp =  await fetchWithToken(`events/${event._id}`, event, "PUT");
      const body = await resp.json();

      if(body.ok) {
        dispatch(eventUpdated(event));
      } else {
        Swal.fire("Error", body.msg, "error");
      }
    } catch (error) {
      console.log(error);
    }
  }
}

export const eventStartDelete = () => {
  return async (dispatch, getState) => {
    const { _id } = getState().calendar.activeEvent;

    try {
      const resp =  await fetchWithToken(`events/${_id}`, {}, "DELETE");
      const body = await resp.json();

      if(body.ok) {
        dispatch(eventDeleted());
      } else {
        Swal.fire("Error", body.msg, "error");
      }
    } catch (error) {
      console.log(error);
    }
  }
}

// sync actions
export const eventAddNew = (event) => ({
  type: types.eventAddNew,
  payload: event,
});

export const eventSetActive = (event) => ({
  type: types.eventSetActive,
  payload: event,
});

export const eventClearActive = () => ({
  type: types.eventClearActive,
});

export const eventUpdated = (event) => ({
  type: types.eventUpdated,
  payload: event,
});

export const eventDeleted = () => ({
  type: types.eventDeleted,
});

export const eventLoad = (events) => ({
  type: types.eventLoad,
  payload: events,
}) 

export const eventLogout = () => ({
  type: types.eventLogout,
});
