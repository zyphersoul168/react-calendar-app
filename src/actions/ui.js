import { types } from "../actionTypes/types";

export const uiOpenModal = () => ({
  type: types.uiOpenModal,
});

export const uiCloseModal = () => ({
  type: types.uiCloseModal,
});
