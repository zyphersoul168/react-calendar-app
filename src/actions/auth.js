import Swal from "sweetalert2";
import { types } from "../actionTypes/types";
import { fetchWithoutToken, fetchWithToken } from "../helpers/fetch";
import { eventLogout } from "./events";

export const startLogin = (email, password) => {
  return async (dispatch) => {
    const response = await fetchWithoutToken(
      "auth/login",
      { email, password },
      "POST"
    );
    const body = await response.json();

    if (body.ok) {
      localStorage.setItem("token", body.token);
      localStorage.setItem("token-init-date", new Date().getTime());

      dispatch(login({ uid: body.uid, name: body.name }));
    } else {
      Swal.fire("Error", body.msg, "error");
    }
  };
};

export const startSignup = (name, email, password) => {
  return async (dispatch) => {
    const response = await fetchWithoutToken(
      "auth/signup",
      { name, email, password },
      "POST",
    );

    const body = await response.json();

    if(body.ok) {
      localStorage.setItem("token", body.token);
      localStorage.setItem("token-init-date", new Date().getTime());

      dispatch(login({ uid: body.uid, name: body.name }));
    } else {
      Swal.fire("Error", body.msg, "error");
    }
  }
};

export const startChecking = () => {
  return async (dispatch) => {
    const response = await fetchWithToken("auth/resignup");

    const body = await response.json();

    if(body.ok) {
      localStorage.setItem("token", body.token);
      localStorage.setItem("token-init-date", new Date().getTime());

      dispatch(login({ uid: body.uid, name: body.name }));
    } else {
      dispatch(setCheckingFinish());
    }
  }
}

export const startLogout = () => {
  return (dispatch) => {
    localStorage.clear();
    dispatch(eventLogout());
    dispatch(logout());
  };
};

// sync actions
export const setCheckingFinish = () => ({
  type: types.authCheckingFinish,
});

export const login = (user) => ({
  type: types.authLogin,
  payload: user,
}); 

export const logout = () => ({
  type: types.authLogout,
});