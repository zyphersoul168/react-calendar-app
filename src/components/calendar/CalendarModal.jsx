import React, { useEffect, useState } from "react";
import Modal from "react-modal";
import DateTimePicker from "react-datetime-picker";
import { useDispatch, useSelector } from "react-redux";
import { SaveIcon, TrashIcon, BackspaceIcon } from "@heroicons/react/outline";
import moment from "moment";

import { uiCloseModal } from "../../actions/ui";
import {
  eventClearActive,
  eventStartAddNew,
  eventStartUpdate,
} from "../../actions/events";
import { DeleteEventBtn } from "../ui/DeleteEventBtn";

const customStyles = {
  content: {
    top: "50%",
    left: "50%",
    right: "auto",
    bottom: "auto",
  },
};

if(process.env.NODE_ENV !== "test") {
  Modal.setAppElement("#root");
}

const now = moment().minutes(0).seconds(0).add(1, "hours");
const nowPlusOne = now.clone().add(1, "hours");

const initialEvent = {
  title: "",
  notes: "",
  start: now.toDate(),
  end: nowPlusOne.toDate(),
};

export const CalendarModal = () => {
  const [, setStartDate] = useState(now.toDate());
  const [, setEndDate] = useState(nowPlusOne.toDate());
  const [dateValid, setDateValid] = useState(true);
  const [titleValid, setTitleValid] = useState(true);

  const [formValues, setFormValues] = useState(initialEvent);

  const { title, notes, start, end } = formValues;

  const dispatch = useDispatch();
  const { modalOpen } = useSelector((state) => state.ui);
  const { activeEvent } = useSelector((state) => state.calendar);

  useEffect(() => {
    if (activeEvent) {
      setFormValues(activeEvent);
    } else {
      setFormValues(initialEvent);
    }
  }, [activeEvent, setFormValues]);

  const handleInputChange = ({ target }) => {
    setFormValues({
      ...formValues,
      [target.name]: target.value,
    });
  };

  const closeModal = () => {
    dispatch(uiCloseModal());
    dispatch(eventClearActive());
    setFormValues(initialEvent);
  };

  const handleStartDateChange = (event) => {
    setStartDate(event);
    setFormValues({
      ...formValues,
      start: event,
    });
  };

  const handleEndDateChange = (event) => {
    setEndDate(event);
    setFormValues({
      ...formValues,
      end: event,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    const momentStartDate = moment(start);
    const momentEndDate = moment(end);

    if (momentStartDate.isSameOrAfter(momentEndDate)) {
      // return Swal.fire("Error", "End date must be greater than start date", "error");
      return setDateValid(false);
    }

    if (title.trim().length < 2) {
      return setTitleValid(false);
    }
    
    if (activeEvent) {
      dispatch(eventStartUpdate(formValues));
    } else {
      dispatch(eventStartAddNew(formValues));
    }

    setDateValid(true);
    setTitleValid(true);
    closeModal();
  };

  return (
    <Modal
      isOpen={modalOpen}
      // onAfterOpen={afterOpenModal}
      onRequestClose={closeModal}
      style={customStyles}
      closeTimeoutMS={200}
      className="modal bg-gray-800 rounded-md text-gray-50 inline w-3/12y outline-none p-6 space-y-4"
      overlayClassName="bg-dark-lava bg-opacity-50 bottom-0 left-0 right-0 top-0 fixed"
      ariaHideApp={!process.env.NODE_ENV === "test"}
    >
      <h1 className="font-abril text-5xl">
        {activeEvent ? (
          activeEvent.title
        ) : (
          <span>
            New <span className="text-indigo-400">Event</span>
          </span>
        )}
      </h1>
      <hr className="h-0.5 bg-opacity-50" />
      <form className="space-y-3" onSubmit={handleSubmit}>
        <div className="space-y-1">
          <label>Date and hour starting</label>
          <DateTimePicker
            onChange={handleStartDateChange}
            value={start}
            className="bg-gray-700 block w-full p-1.5 rounded-md"
            calendarIcon={<TrashIcon className="w-5 h-5" />}
            clearIcon={<BackspaceIcon className="w-5 h-5" />}
          />
        </div>

        <div className="space-y-1">
          <label>Date and hour ending</label>
          {!dateValid && (
            <span className="date-error block text-red-400">
              Error: End date must be greater than start date
            </span>
          )}
          <DateTimePicker
            onChange={handleEndDateChange}
            value={end}
            minDate={start}
            className="bg-gray-700 block w-full p-1.5 rounded-md"
            calendarIcon={<TrashIcon className="w-5 h-5" />}
            clearIcon={<BackspaceIcon className="w-5 h-5" />}
          />
        </div>
        <hr className="h-0.5 bg-opacity-50" />
        <div className="space-y-1">
          <label>Title and notes</label>
          {!titleValid && (
            <span className="error-span block text-red-400">Error: Title too short</span>
          )}
          <input
            type="text"
            className="bg-gray-700 block w-full p-1.5 rounded-md focus:outline-none"
            placeholder="Event title"
            name="title"
            autoComplete="off"
            onChange={handleInputChange}
            value={title}
          />
        </div>

        <div>
          <small id="emailHelp" className="form-text text-muted">
            A short description
          </small>
          <div className="form-group">
            <textarea
              type="text"
              className="bg-gray-700 block w-full p-1.5 rounded-md focus:outline-none"
              placeholder="Notes"
              rows="5"
              name="notes"
              onChange={handleInputChange}
              value={notes}
            ></textarea>
          </div>
          <small id="emailHelp" className="form-text text-muted">
            Aditional info
          </small>
        </div>

        <button
          type="submit"
          className="hover:bg-green-600 bg-green-400 p-2.5 font-bold rounded-md text-alabaster flex justify-center w-full transition ease-in"
        >
          <SaveIcon className="mt-1 mr-1 h-5 w-5" />
          <span>Save</span>
        </button>
        {activeEvent && <DeleteEventBtn />}
      </form>
    </Modal>
  );
};
