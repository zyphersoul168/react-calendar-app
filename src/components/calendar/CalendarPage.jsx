import React, { useEffect, useState } from "react";
import { Calendar, momentLocalizer } from "react-big-calendar";
import { useDispatch, useSelector } from "react-redux";
import moment from "moment";

import { Navbar } from "../ui/Navbar";
import { CalendarEvent } from "./CalendarEvent";
import { CalendarModal } from "./CalendarModal";
import { uiOpenModal } from "../../actions/ui";

import "react-big-calendar/lib/css/react-big-calendar.css";
import {
  eventSetActive,
  eventStartLoading,
} from "../../actions/events";
import { AddNewFab } from "../ui/AddNewFab";

const localizer = momentLocalizer(moment);

export const CalendarPage = () => {
  const [lastView, setLastView] = useState(
    localStorage.getItem("lastView") || "month"
  );
  const dispatch = useDispatch();
  const { events } = useSelector((state) => state.calendar);
  const { uid } = useSelector((state) => state.auth);

  useEffect(() => {
    dispatch(eventStartLoading());
  }, [dispatch]);

  const onDoubleClick = (event) => {
    dispatch(uiOpenModal());
  };

  const onSelectEvent = (event) => {
    dispatch(eventSetActive(event));
  };

  const onViewChange = (event) => {
    setLastView(event);
    localStorage.setItem("lastView", event);
  };

  // const onSelectSlot = (event) => {
  //   dispatch(eventClearActive());

  //   if (event.action === "doubleClick") {
  //     dispatch(uiOpenModal());
  //     dispatch(
  //       eventSetActive({
  //         id: new Date().getDate(),
  //         title: "",
  //         start: event.start,
  //         end: event.end,
  //         bgcolor: "#212121",
  //         notes: "",
  //         user: {
  //           _id: "123",
  //           name: "aaron",
  //         },
  //       })
  //     );
  //   }
  // };

  const eventStyleGetter = (event, start, end, isSelected) => {
    const style = {
      backgroundColor: uid === event.user._id ? "#e0afa0" : "#463f3a",
      borderWidth: 0,
      color: "#f4f3ee",
      display: "block",
    };

    return {
      style,
    };
  };

  return (
    <div>
      <Navbar />
      <div className="mt-6">
        <AddNewFab />
        <Calendar
          localizer={localizer}
          events={events}
          startAccessor="start"
          endAccessor="end"
          style={{ height: "92vh" }}
          eventPropGetter={eventStyleGetter}
          onDoubleClickEvent={onDoubleClick}
          onSelectEvent={onSelectEvent}
          onView={onViewChange}
          // onSelectSlot={onSelectSlot}
          // selectable={true}
          view={lastView}
          components={{
            event: CalendarEvent,
          }}
        />
      </div>
      <CalendarModal />
    </div>
  );
};
