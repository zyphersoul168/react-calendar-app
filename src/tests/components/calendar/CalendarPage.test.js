import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import "@testing-library/jest-dom";
import { act } from "@testing-library/react";

import { CalendarPage } from "../../../components/calendar/CalendarPage";
import { types } from "../../../actionTypes/types";
import { eventSetActive } from "../../../actions/events";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {
  calendar: {
    events: [],
  },
  auth: {
    uid: "1234",
    name: "aaron",
  },
  ui: {
    modalOpen: false,
  },
};
const store = mockStore(initialState);

// mocks
Storage.prototype.setItem = jest.fn();

store.dispatch = jest.fn();

jest.mock("../../../actions/events", () => ({
  eventSetActive: jest.fn(),
  eventStartLoading: jest.fn(),
}));

const wrapper = mount(
  <Provider store={store}>
    <CalendarPage />
  </Provider>
);

describe("Tests on CalendarPage component", () => {
  test("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should dispatch calendar actions", () => {
    const calendar = wrapper.find("Calendar");

    calendar.prop("onDoubleClickEvent")();
    expect(store.dispatch).toHaveBeenCalledWith({
      type: types.uiOpenModal,
    });

    calendar.prop("onSelectEvent")({ start: "Hola" });
    expect(eventSetActive).toHaveBeenCalledWith({ start: "Hola" });

    act(() => {
      calendar.prop("onView")("week");
      expect(localStorage.setItem).toHaveBeenCalledWith("lastView", "week");
    });
  });
});
