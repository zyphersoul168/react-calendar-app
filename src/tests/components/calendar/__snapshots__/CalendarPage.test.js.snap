// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on CalendarPage component should match snapshot 1`] = `
<div
  className="calendar-screen"
>
  <nav
    className="bg-dark-lava shadow-lg"
  >
    <div
      className="max-w-10xl mx-auto px-4"
    >
      <div
        className="flex justify-between"
      >
        <div
          className="flex space-x-7"
        >
          <div
            className="hidden md:flex items-center space-x-4"
          >
            <p
              className="py-4 px-2 text-alabaster"
            >
              aaron
            </p>
          </div>
        </div>
        <div
          className="hidden md:flex items-center space-x-3 "
        >
          <button
            className="inline-flex py-2 px-2 font-bold text-alabaster bg-melon rounded transition duration-300"
            onClick={[Function]}
          >
            <svg
              className="mt-0.5 mr-1 h-5 w-5"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 16l4-4m0 0l-4-4m4 4H7m6 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h4a3 3 0 013 3v1"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
              />
            </svg>
             Logout
          </button>
        </div>
        <div
          className="md:hidden flex items-center"
        >
          <button
            className="outline-none mobile-menu-button"
          >
            <svg
              className=" w-6 h-6 text-gray-500 hover:text-green-500 "
              fill="none"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              x-show="!showMenu"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      className="hidden mobile-menu"
    >
      <ul
        className=""
      >
        <li
          className="active"
        >
          <a
            className="block text-sm px-2 py-4 text-white bg-green-500 font-semibold"
            href="index.html"
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="block text-sm px-2 py-4 hover:bg-green-500 transition duration-300"
            href="#services"
          >
            Services
          </a>
        </li>
        <li>
          <a
            className="block text-sm px-2 py-4 hover:bg-green-500 transition duration-300"
            href="#about"
          >
            About
          </a>
        </li>
        <li>
          <a
            className="block text-sm px-2 py-4 hover:bg-green-500 transition duration-300"
            href="#contact"
          >
            Contact Us
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="mt-6"
  >
    <button
      className="bg-melon p-4 absolute bottom-3 right-3 rounded-full z-10 text-dark-lava"
      onClick={[Function]}
    >
      <svg
        className="w-6 h-6"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10 3a1 1 0 011 1v5h5a1 1 0 110 2h-5v5a1 1 0 11-2 0v-5H4a1 1 0 110-2h5V4a1 1 0 011-1z"
          fillRule="evenodd"
        />
      </svg>
    </button>
    <div
      className="rbc-calendar"
      style={
        Object {
          "height": "92vh",
        }
      }
    >
      <div
        className="rbc-toolbar"
      >
        <span
          className="rbc-btn-group"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Today
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Back
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </span>
        <span
          className="rbc-toolbar-label"
        >
          October 2021
        </span>
        <span
          className="rbc-btn-group"
        >
          <button
            className="rbc-active"
            key="month"
            onClick={[Function]}
            type="button"
          >
            Month
          </button>
          <button
            className=""
            key="week"
            onClick={[Function]}
            type="button"
          >
            Week
          </button>
          <button
            className=""
            key="day"
            onClick={[Function]}
            type="button"
          >
            Day
          </button>
          <button
            className=""
            key="agenda"
            onClick={[Function]}
            type="button"
          >
            Agenda
          </button>
        </span>
      </div>
      <div
        aria-label="Month View"
        className="rbc-month-view"
        role="table"
      >
        <div
          className="rbc-row rbc-month-header"
          role="row"
        >
          <div
            className="rbc-header"
            key="header_0"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Sun
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_1"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Mon
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_2"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Tue
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_3"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Wed
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_4"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Thu
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_5"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Fri
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_6"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              Sat
            </span>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  26
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  28
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  02
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  03
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  04
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  05
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  06
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  07
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  08
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  09
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  10
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  11
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  12
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  13
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  14
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  15
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  16
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg rbc-today"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell rbc-now rbc-current"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  17
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  18
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  19
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  20
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  21
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  22
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  23
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  24
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  25
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  26
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  28
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  30
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  31
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_1"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_2"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  02
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_3"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  03
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_4"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  04
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_5"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  05
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_6"
                role="cell"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  role="cell"
                >
                  06
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <Portal
    containerInfo={
      <div
        class="ReactModalPortal"
      />
    }
  />
</div>
`;
